src/__experimental__/components/decimal/decimal.spec.js

```diff
@@ -779,16 +765,34 @@ describe("Decimal", () => {
 
         it("does not format a value with numbers and too many separators", () => {
           const onBlur = jest.fn();
-          render({ onBlur, precision: 4 });
+          render({ onBlur, precision: 4, ...itProps });
           type("1,,2345,56");
           blur();
           expect(value()).toBe("1,,2345,56");
           expect(hiddenValue()).toBe("1,,2345,56");
         });
+
+        it("supports having a shorter number", () => {
+          render({ defaultValue: "12345.6", precision: 3, ...itProps });
+          expect(value()).toBe("12.345,600");
+          expect(hiddenValue()).toBe("12345.600");
+        });
+
+        it("supports having a longer number on default", () => {
+          render({ defaultValue: "123123.99999", precision: 3, ...itProps });
+          expect(value()).toBe("123.123,99999");
+          expect(hiddenValue()).toBe("123123.99999");
+        });
+
+        it("supports having a longer number on change", () => {
```

Comment:
This test has a different description but looks exactly the same as the test above?

Reply:
It's not the same case - defaultValue vs value - second test is for changing after defaultValue is not passed.